### PR TITLE
Flow BM config overrides

### PR DESCRIPTION
**Pull request by amitdahan**
### 🔦 Summary
Closes https://github.com/wix/yoshi/issues/2265.

Allows overriding the `moduleId` (through `<root>/yoshi-bm.config.json`) and `componentId` (through i.e `src/pages/index.config.json`).

This is still unvalidated & un-cosmiconfig-ified, for now.

### 🗺️ Test Plan
Fixed template's config file to be `yoshi-bm.config.json`, rest should work the same.

### CODE REVIEW

File: packages/yoshi-flow-bm/src/createFlowBMModel.ts [hunk 1]
```diff
@@ -44,38 +49,56 @@ export default function createFlowBMModel(cwd = process.cwd()): FlowBMModel {
...
+  const [localePath] = globDirs(translationsDir);
 
   return {
-    moduleId,
+    moduleId: config.moduleId,
     pages,
     exportedComponents,
     methods,
     localePath,
-    moduleConfig: {},
```
Comment (amitdahan) [inline packages/yoshi-flow-bm/src/createFlowBMModel.ts:78]: Unused, removed for now.

File: packages/yoshi-flow-bm/src/config.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import path from 'path';
+import fs from 'fs-extra';
+import { getProjectArtifactId } from 'yoshi-helpers/build/utils';
+
+export interface FlowBMConfig {
+  moduleId: string;
+  routeNamespace: string;
+}
+
+const configPath = 'yoshi-bm.config.json';
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/config.ts:10]: Put in constants

File: packages/yoshi-flow-bm/src/config.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+function readJson(jsonPath: string): Record<string, any> {
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/config.ts:12]: readJsonSilent + put in utils

File: packages/yoshi-flow-bm/src/config.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  return !fs.existsSync(jsonPath) ? {} : fs.readJSONSync(jsonPath);
+}
+
+export function readConfig(cwd = process.cwd()): FlowBMConfig {
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/config.ts:16]: remove default here

File: packages/yoshi-flow-bm/src/config.ts [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+  return {
+    moduleId: getProjectArtifactId(cwd)!,
+    routeNamespace: '',
+    ...readJson(path.resolve(cwd, configPath)),
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/config.ts:20]: move to a variable

File: packages/yoshi-flow-bm/src/config.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+export function readPageConfig(
+  { moduleId }: FlowBMConfig,
+  pagePath: string,
+): PageConfig {
+  const { dir, name } = path.parse(pagePath);
+
+  return {
+    componentId: `${moduleId}.pages.${path.parse(pagePath).name}`,
+    ...readJson(path.join(dir, `${name}.config.json`)),
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/config.ts:36]: put in variable

File: packages/yoshi-flow-bm/src/createFlowBMModel.ts [hunk 7]
```diff
@@ -22,15 +27,15 @@ export interface FlowBMModel {
...
 
 const exts = '{js,jsx,ts,tsx}';
-const pagesPattern = `src/pages/**/*.${exts}`;
+const pagesDir = 'src/pages';
+const pagesPattern = `${pagesDir}/**/*.${exts}`;
 const componentsPattern = `src/exported-components/**/*.${exts}`;
 const methodsPattern = `src/methods/**/*.${exts}`;
 const moduleInitPattern = `src/moduleInit.${exts}`;
-const translationsPattern = 'translations';
+const translationsDir = 'translations';
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/createFlowBMModel.ts:38]: CONSTANTS

File: packages/yoshi-flow-bm/src/createFlowBMModel.ts [hunk 8]
```diff
@@ -44,38 +49,56 @@ export default function createFlowBMModel(cwd = process.cwd()): FlowBMModel {
       expandDirectories: false,
     });
 
-  const moduleId = getProjectArtifactId(cwd)!;
+  const config = readConfig(cwd);
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/createFlowBMModel.ts:52]: readModuleConfig (`.module.json`)

File: packages/yoshi-flow-bm/src/createFlowBMModel.ts [hunk 9]
```diff
@@ -44,38 +49,56 @@ export default function createFlowBMModel(cwd = process.cwd()): FlowBMModel {
...
+    return {
+      ...readPageConfig(config, pagePath),
+      componentPath: pagePath,
+      route: path.join(
+        config.routeNamespace,
+        ...path
+          .relative(pagesDir, pagePath)
+          .split(path.delimiter)
+          .slice(0, -1),
+        name !== 'index' ? name : '',
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/createFlowBMModel.ts:66]: put to variable

File: packages/yoshi-flow-bm/src/createFlowBMModel.ts [hunk 10]
```diff
@@ -44,38 +49,56 @@ export default function createFlowBMModel(cwd = process.cwd()): FlowBMModel {
...
+    ...readExportedComponentConfig(config, componentPath),
+    componentPath,
+  });
+
+  const getMethodModel = (methodPath: string): MethodModel => ({
+    ...readMethodConfig(config, methodPath),
+    methodPath,
+  });
+
+  const pagesDir = path.join(cwd, 'src/pages');
```
Comment (ranyitz) [inline packages/yoshi-flow-bm/src/createFlowBMModel.ts:83]: `src/pages` to constant

File: packages/yoshi-helpers/src/utils.ts [hunk 11]
```diff
@@ -1,6 +1,7 @@
 import fs from 'fs';
 import path from 'path';
 import childProcess, { ChildProcess } from 'child_process';
+import fsX from 'fs-extra';
```
Comment (ranyitz) [inline packages/yoshi-helpers/src/utils.ts:4]: This is not needed, `fs-extra` is a superset of `fs`. See docs - https://github.com/jprichardson/node-fs-extra#nodejs-fs-extra

Comment (amitdahan) [reply packages/yoshi-helpers/src/utils.ts:4]: Cool, was kinda scared to break unrelated things but sounds good!

**Pull request merged**